src/apps/dashboard/features/plugins/components/NewRepositoryForm.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useCallback } from 'react';
+import type { RepositoryInfo } from '@jellyfin/sdk/lib/generated-client/models/repository-info';
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+import globalize from 'lib/globalize';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import Button from '@mui/material/Button';
+import Stack from '@mui/material/Stack';
+import TextField from '@mui/material/TextField';
+
+type IProps = {
+    open: boolean;
+    onClose: () => void;
+    onAdd: (repository: RepositoryInfo) => void;
+};
+
+const NewRepositoryForm = ({ open, onClose, onAdd }: IProps) => {
+    const onSubmit = useCallback((e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+
+        const formData = new FormData(e.currentTarget);
+        const data = Object.fromEntries(formData.entries());
+
+        const repository: RepositoryInfo = {
+            'Name': data.Name?.toString(),
+            'Url': data.Url?.toString(),
+            'Enabled': true
```

Comment:
Quotes aren't needed here right?

```suggestion
            Name: data.Name?.toString(),
            Url: data.Url?.toString(),
            Enabled: true
```